lib/dealership.rb

```diff
@@ -0,0 +1,54 @@
+class Dealership
+  attr_reader :dealer_name, :address, :inventory
+  def initialize(dealer_name, address)
+    @dealer_name = dealer_name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    @inventory.length > 0
+  end
+
+  def cars_by_make(make)
+    @inventory.select {|car_make| car_make.make == make}
+  end
+
+  def total_value
+    # Grabs the total cost of each car and adds
+    # the total value of all of them
+    total = @inventory.map {|value| value.total_cost}
+    total.reduce(:+)
+  end
+
+  def details
+    hash = {}
+    t_value = total_value
+    dealer_address = @address
+    hash["total_value"] = t_value
+    hash["address"] = dealer_address
+    hash
```

Comment:
another way you could simplify this a bit could be:

```
hash = {}
hash["total_value"] = total_value
hash["address"] = @address
hash
```

rather than creating additional local variables.

An even shorter syntax could be:

```
{ "total_value" => total_value, "address" => @address }
```